site/app/models/notebook/UserSpecificNotebook.php

```diff
@@ -109,7 +109,19 @@ private function replaceNotebookItemsWithQuestions(array $raw_notebook): array {
      */
     private function getItemFromPool(array $item): string {
         $item_label = $item['item_label'];
-        $selected = $this->getNotebookHash($item_label, count($item['from_pool']));
+        $selected = null;
+        // Check if user-mapping is available or not
+        if (
```

Comment:
Instead of this check you can do
` $selected = $item['user_item_map'][$this->user_id] ?? null`;

Since a similar check is performed in the C++ side and if the value does somehow become invalud/out of bounds you can fall back to using the hash by setting it null

Reply:
are you suggesting to remove this conditional checking whether the user index is set and value lies in the valid range?

Reply:
yeah since the null coalescing operater will check if the value user_item_map and this->user_id is set and not null so you can avoid the check, if the value is out of range and note there it will default to null and then fallback to the normal hashing method. I think its a cleaner way than the large if statement

Reply:
Yeah, make sense since I'm already performing the rane validation check in CPP code, thanks:)